src/integrations/gitea/create_release.rs

```diff
@@ -0,0 +1,129 @@
+use std::io::Write;
+
+use miette::Diagnostic;
+
+use super::initialize_state;
+use crate::{
+    app_config, config,
+    dry_run::DryRun,
+    integrations::{ureq_err_to_string, CreateReleaseInput, CreateReleaseResponse},
+    state,
+    step::releases::package::{Asset, AssetNameError},
+};
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn create_release(
+    name: &str,
+    tag_name: &str,
+    body: Option<&str>,
+    prerelease: bool,
+    gitea_state: state::Gitea,
+    gitea_config: &config::Gitea,
+    dry_run_stdout: DryRun,
+    assets: Option<&Vec<Asset>>,
+) -> Result<state::Gitea, Error> {
+    let gitea_release = CreateReleaseInput::new(tag_name, name, body, prerelease, false);
+
+    if let Some(stdout) = dry_run_stdout {
+        gitea_release_dry_run(name, assets, &gitea_release, stdout)?;
+        return Ok(gitea_state);
+    }
+
+    let (token, agent) = initialize_state(&gitea_config.host, gitea_state)?;
+
+    agent
+        .post(&gitea_config.get_releases_url())
+        .query("access_token", &token)
+        .send_json(gitea_release)
+        .map_err(|source| Error::ApiRequest {
+            err: ureq_err_to_string(source),
+            activity: "creating a release".to_string(),
+            host: gitea_config.host.clone(),
+        })?
+        .into_json::<CreateReleaseResponse>()
+        .map_err(|source| Error::ApiResponse {
+            source,
+            activity: "creating a release",
+            host: gitea_config.host.clone(),
+        })?;
+
+    /*
+     * NOTE: ureq currently doesn't support support multipart/form-data
+     *       so we cannot support release attachments on any Gitea instance
+     *       see: https://github.com/algesten/ureq/issues/698
+     *       docs: https://try.gitea.io/api/swagger#/repository/repoCreateReleaseAttachment
+     */
+    if assets.is_some() {
+        println!("NOTE: adding assets to a gitea release is currently not supported. Uploading assets will be skipped");
+    }
```

Comment:
Oof. Maybe it's finally time to accept the fact that I'll need to pull in an async runtime. If we support pushing to multiple forges at once, there might finally be a parallelism benefit anyway.

Reply:
Maybe we should error at the config level if both assets and `gitea` are configured, so someone doesn't find out post-release after digging through CI logs 😦

Reply:
> Maybe we should error at the config level if both assets and `gitea` are configured, so someone doesn't find out post-release after digging through CI logs 😦

yeah, seems like a good idea to me, It's how i figured it out ^^

> Oof. Maybe it's finally time to accept the fact that I'll need to pull in an async runtime. If we support pushing to multiple forges at once, there might finally be a parallelism benefit anyway.

If you do decide to do so, i'd recommend the `reqwest` crate. been using it for pretty much everything that needs to request stuff.